### PR TITLE
fix: consider element's accessible names when labels are hidden

### LINKED ISSUE #1176: Browser extension: axe is ignoring hidden labels for input fields even when it should acknowledge them
There are common instances where we often require labels of input fields to be hidden (e.g. Search box because it already has a search icon in it).

As per [this W3C article](https://www.w3.org/WAI/tutorials/forms/labels/#note-on-hiding-elements), the recommended way to achieve this is to add the below CSS code to the label:
<pre>.visuallyhidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}</pre>

But even after implementing this, aXe still throws an error for the field saying `Form elements must have labels`.

Here is a working demo of the same: https://jsfiddle.net/xk2vaq6f/1/embedded/result

### PR DESCRIPTION
**Pull request by stephenmathieson**
Closes #1176

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [X] Follows the commit message policy, appropriate for next version
- [X] Has documentation updated, a DU ticket, or requires no documentation change
- [X] Includes new tests, or was unnecessary
- [X] Code is reviewed for security by: Marcy Sutton

### CODE REVIEW

File: lib/checks/label/hidden-explicit-label.js [hunk 1]
```diff
@@ -2,9 +2,12 @@ if (node.getAttribute('id')) {
 	const root = axe.commons.dom.getRootNode(node);
 	const id = axe.commons.utils.escapeSelector(node.getAttribute('id'));
 	const label = root.querySelector(`label[for="${id}"]`);
+	const isLabelVisible = label && axe.commons.dom.isVisible(label);
 
-	if (label && !axe.commons.dom.isVisible(label)) {
-		return true;
+	if (isLabelVisible === false) {
```
Comment (WilcoFiers) [inline lib/checks/label/hidden-explicit-label.js]: I think this now fails if there is no label. Should probably just put the if statement back to the way it was.

Comment (stephenmathieson) [reply lib/checks/label/hidden-explicit-label.js]: Good catch. Will update!

File: lib/checks/label/hidden-explicit-label.js [hunk 2]
```diff
@@ -4,7 +4,9 @@ if (node.getAttribute('id')) {
 	const label = root.querySelector(`label[for="${id}"]`);
 
 	if (label && !axe.commons.dom.isVisible(label)) {
```
Comment (WilcoFiers) [inline lib/checks/label/hidden-explicit-label.js]: We should change this so it does `!axe.commons.dom.isVisible(label, true))`.

Here's a test case for it:

```js
it('should fail when the label has aria-hidden=true', function() {
	var html = '';
	html += '<div>';
	html += '  <label for="target" aria-hidden="true">';
	html += '    Hello world';
	html += '  </label>';
	html += '  <input id="target">';
	html += '</div>';
	var args = checkSetup(html, {}, '#target');
	assert.isTrue(check.evaluate.apply(check, args));
});
```

Comment (stephenmathieson) [reply lib/checks/label/hidden-explicit-label.js]: Good call! Updated.

**Pull request merged**